dev/benchmarking/ddl/1_tables.sql

```diff
@@ -0,0 +1,291 @@
+CREATE TABLE IF NOT EXISTS public.project
+(
+  id SERIAL PRIMARY KEY
+  , project_name citext NOT NULL
+  , project_url text NOT NULL
+  , repo_url text NOT NULL
+  , last_changed timestamp (0) without time zone NOT NULL DEFAULT now()
+);
+COMMENT ON TABLE public.project
+  IS 'Project name and relevant URLs.';
+COMMENT ON COLUMN public.project.project_url
+  IS 'Homepage URL.';
+COMMENT ON COLUMN public.project.repo_url
+  IS 'Git repo URL to link stored commit hashes to code in a webpage.';
+COMMENT ON COLUMN public.project.last_changed
+  IS 'New project details are added with a new timestamp. '
+     'The project details with the newest timestamp will be used.';
+
+-- BENCHMARK_LANGUAGE
+CREATE TABLE IF NOT EXISTS public.benchmark_language
+(
+  id SERIAL PRIMARY KEY
+  , benchmark_language citext NOT NULL UNIQUE
+);
+COMMENT ON TABLE public.benchmark_language
+  IS 'The language the benchmark was written in (and presumably for).';
+COMMENT ON COLUMN public.benchmark_language.benchmark_language
+  IS 'The benchmark language. For example: Python';
+
+-- LANGUAGE_DEPENDENCIES
+CREATE TABLE IF NOT EXISTS public.language_dependencies
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , dependencies jsonb NOT NULL DEFAULT '{}'::jsonb
+  , PRIMARY KEY (id, benchmark_language_id)
+  , FOREIGN KEY (benchmark_language_id) REFERENCES public.benchmark_language
+);
+COMMENT ON TABLE public.language_dependencies
+  IS E'@name language_dependencies.\n'
+      'A JSON string mapping dependencies to their versions for each language.';
+COMMENT ON COLUMN public.language_dependencies.dependencies
+  IS 'For example: ''{"boost": "1.69", "conda": "", "numpy": "1.15"}''.';
+
+-- LANGUAGE_VERSION
+CREATE TABLE IF NOT EXISTS public.language_version
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , language_version citext NOT NULL DEFAULT ''
+  , PRIMARY KEY (id, benchmark_language_id)
+  , FOREIGN KEY (benchmark_language_id) REFERENCES public.benchmark_language
+);
+COMMENT ON TABLE public.language_version
+  IS 'The benchmark language version, e.g. ''2.7'' for Python 2.7';
+COMMENT ON COLUMN public.language_version.language_version
+  IS 'The version number used in the benchmark environment (e.g. ''2.7'').';
+
+-- BENCHMARK_TYPE
+CREATE TABLE IF NOT EXISTS public.benchmark_type
+(
+  id SERIAL PRIMARY KEY
+  , benchmark_type citext NOT NULL UNIQUE
+);
+COMMENT ON TABLE public.benchmark_type
+  IS 'The type of benchmark. For example "time", "mem", "peakmem", "track"';
+COMMENT ON COLUMN public.benchmark_type.benchmark_type
+  IS 'The type of units, so "time" for sec/ns/ms or "mem" for kB, MB, GB.';
+
+-- MACHINE
+CREATE TABLE IF NOT EXISTS public.machine
+(
+  id SERIAL PRIMARY KEY
+  , machine_name citext NOT NULL UNIQUE
+  , memory_bytes bigint NOT NULL
+  , cpu_actual_frequency_Hz bigint NOT NULL
+  , cpu_id integer NOT NULL
+  , gpu_id integer NOT NULL
+  , os_id integer NOT NULL
+  , FOREIGN KEY (cpu_id) REFERENCES public.cpu
```

Comment:
might be useful to have key-value pairs here to describe other attributes until they can be incorporated more fully into the schema (e.g. hard disk type)

Reply:
also not hard to add. will do. Thank you!